client/question/components/LearnerViewDirective.js

```diff
@@ -1180,13 +1182,46 @@ tie.directive('learnerView', [function() {
             tasks[currentTaskIndex].getId());
         };
 
+        /**
+         * Event handler to hide modal after transition animation ends.
+         */
+        $scope.hideModalAfterAnimation = function(event) {
+          $timeout(function() {
+            MonospaceDisplayModalService.hideModal();
+          }, 0);
+          event.target.removeEventListener("transitionend",
+              $scope.hideModalAfterAnimation, false);
+        };
+
+        /**
+         *  Close modal by calling MonospaceDisplayModalService.closeModal.
+         */
+        $scope.closeModal = function() {
```

Comment:
The code here duplicates code in MonospaceDisplayModalDirective. In general try to strictly avoid duplicating code since that makes things harder to maintain and introduces the possibility of skew.